.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,150 @@
+  workflow_call:
+    inputs:
+      python_v:
+        description: "Python version needed to run the code, eg. 3.10.6"
+        default: 3.10.6
+        required: false
+        type: string
+      source_dir:
+        description: "The source directory of the code, to be used by Sonar Scan"
+        required: false
+        type: string
+      exclude_from_coverage:
+        description: "Directories to be excluded from the test coverage report"
+        required: false
+        type: string
+
+env:
+  PERSONAL_ACCESS_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+
+jobs:
+  # We need the latest release for dev, beta and stable to define the new code for Sonarcloud
+  # If no previous release available, the overall code will be analyzed.
+  # (New code = code changes from the latest release)
+  latestRelease:
+    name: Get latest release
+    runs-on: ubuntu-20.04
+    continue-on-error: true
+    outputs:
+      latest_release: ${{ steps.latestRelease.outputs.tag }}
+    steps:
+      - name: Get latest dev tag
+        if: github.ref == 'refs/heads/dev' && github.event_name != 'pull_request'
+        uses: oprypin/find-latest-tag@v1
+        with:
+          repository: ${{ github.repository }}
+          releases-only: true
+          regex: 'v\d+\.\d+\.\d+a\d+$'
+          token: ${{ env.PERSONAL_ACCESS_TOKEN }}
+        id: latestDevTag
+      - name: Get latest beta tag
+        if: github.ref == 'refs/heads/beta' && github.event_name != 'pull_request'
+        uses: oprypin/find-latest-tag@v1
+        with:
+          repository: ${{ github.repository }}
+          releases-only: true
+          regex: 'v\d+\.\d+\.\d+b\d+$'
+          token: ${{ env.PERSONAL_ACCESS_TOKEN }}
+        id: latestBetaTag
+      - name: Get latest stable tag
+        if: github.ref == 'refs/heads/stable' && github.event_name != 'pull_request'
+        uses: oprypin/find-latest-tag@v1
+        with:
+          repository: ${{ github.repository }}
+          releases-only: true
+          regex: 'v\d+\.\d+\.\d+(.post\d+)?$'
+          token: ${{ env.PERSONAL_ACCESS_TOKEN }}
+        id: latestStableTag
+      - name: Select latest version for ${{ github.ref }}
+        run: |
+          if [ ${{ steps.latestDevTag.outputs.tag }} ]; then
+            echo 'tag=${{ steps.latestDevTag.outputs.tag }}' >> $GITHUB_OUTPUT
+          elif [ ${{ steps.latestBetaTag.outputs.tag }} ]; then
+            echo 'tag=${{ steps.latestBetaTag.outputs.tag }}' >> $GITHUB_OUTPUT
+          elif [ ${{ steps.latestStableTag.outputs.tag }} ]; then
+            echo 'tag=${{ steps.latestStableTag.outputs.tag }}' >> $GITHUB_OUTPUT
+          else
+            echo 'tag=' >> $GITHUB_OUTPUT
+          fi
+        id: latestRelease
+      - run: echo "New code defined since version ${{ steps.latestRelease.outputs.tag }} (only on push triggers)"
+
+  build:
+    name: Node.js CI
+    runs-on: ubuntu-20.04
+    needs: latestRelease
+    strategy:
+      matrix:
+        node-version: [19.x]
```

Comment:
Maybe we could add the node version as an input parameter, instead of python_v. We may have different projects using the workflow in the future, that need different node versions

Reply:
done